packages/adapter-drizzle/src/drivers/mysql.ts

```diff
@@ -31,11 +31,19 @@ export class DrizzleMySQLAdapter implements Adapter {
 	public async getSessionAndUser(
 		sessionId: string
 	): Promise<[session: DatabaseSession | null, user: DatabaseUser | null]> {
-		const [databaseSession, databaseUser] = await Promise.all([
-			this.getSession(sessionId),
-			this.getUserFromSessionId(sessionId)
-		]);
-		return [databaseSession, databaseUser];
+		const result = await this.db
+			.select({
+				user: this.userTable,
+				session: this.sessionTable
+			})
+			.from(this.sessionTable)
+			.innerJoin(this.userTable, eq(this.sessionTable.userId, this.userTable.id))
+			.where(eq(this.sessionTable.id, sessionId));
+		if (result.length !== 1) return [null, null];
+		return [
+			transformIntoDatabaseSession(result[0].session),
```

Comment:
Will `result[0].session` be `null` if for some reason the session/user didn't exist?

Reply:
If session/user doesn't exist, result will be an empty array.

Reply:
I mean like if the session exists but the user doesn’t

Reply:
It will throw FOREIGN KEY constraint error (if a relation exists), otherwise just empty array.